src/lib/ingest.js

```diff
@@ -16,27 +16,26 @@ export class InvalidIngestError extends Error {
 
 const hierarchyLinks = ['self', 'root', 'parent', 'child', 'collection', 'item', 'items']
 
-export async function convertIngestObjectToDbObject(
-  // eslint-disable-next-line max-len
-  /** @type {{ hasOwnProperty: (arg0: string) => any; type: string, collection: string; links: any[]; id: any; }} */ data
-) {
-  let index = ''
+export async function convertIngestMsgToDbOperation(data) {
+  let index
+  let action
   logger.debug('data', data)
   if (isCollection(data)) {
     index = COLLECTIONS_INDEX
+    action = 'index'
   } else if (isItem(data)) {
     index = data.collection
+    action = 'index'
+  } else if (isAction(data)) {
+    index = data.collection
+    action = data.command
   } else {
     throw new InvalidIngestError(
-      `Expected data.type to be 'Collection' or 'Feature' not '${data.type}'`
+      `Expected 'type' to be 'Collection', 'Feature', or 'action', not '${data.type}'`
     )
   }
 
-  // remove any hierarchy links in a non-mutating way
-  if (!data.links) {
-    throw new InvalidIngestError('Expected a "links" property on the stac object')
-  }
-  const links = data.links.filter(
+  const links = (data.links || []).filter(
```

Comment:
For whatever reason, ingest rejected the Item if it was missing a `links` field with an array value. While this is required for a valid STAC Item, it's one of the only validations that ingest actually does, so it's pretty useless. This just adds an empty links array if it's missing.